plugins/easyimage/plugin.js

```diff
@@ -148,8 +152,10 @@
 
 					if ( data.type === 'side' ) {
 						this.element.addClass( editor.config.easyimage_sideClass );
+						this.wrapper.addClass( editor.config.easyimage_sideClass );
```

Comment:
By adding exactly same class to the wrapper you're applying the styling twice, once to the wrapper, and then again to the widget element itself.

In this case you used `max-width: 50%`  but the image can never get to 50% (as stated in manual test expectation). See below screen:

![Image set to 25% width of the editor with other 25% consumed by blank padding](https://user-images.githubusercontent.com/5353898/32694573-f77df866-c743-11e7-906d-1a8e6445d6d4.png)

Reply:
Second thing is that you could avoid this kind of code duplication by calling [`widget.addClass()`](http://docs.ckeditor.test/#!/api/CKEDITOR.plugins.widget-method-addClass) (and `widget.removeClass` respectively) rather than doing this stuff manually.

By doing so your wrapper gets applied class prefixed with `cke_widget_wrapper_`. So you could fix above issue with CSS like that:

```css
.easyimage-side, .cke_widget_wrapper_easyimage-side {
	float: right;
	max-width: 25%;
}

.cke_widget_wrapper_easyimage-side > .easyimage-side {
	float: none;
	max-width: none;
}
```

Although this syntax still looks like a garbage to me (but it's a known fact I'm nor a CSS expert or a CSS lover).

Reply:
Our API still could surprise me 😅

Reply:
Actually your styles could be simplified to:

```css
.easyimage-side, .cke_widget_wrapper_easyimage-side {
	float: right;
	max-width: 25%;
}
```

Reply:
The reason why I cared for `.easyimage-side` (with no wrapper) alone was that I wanted this stylesheet to be compatible with the content created by CKEditor. So that if someone toggles CKEditor in and out it still is styled nicely using same file.

To see it play with [/tests/plugins/easyimage/manual/styles](http://tests.ckeditor.test:1030/tests/plugins/easyimage/manual/styles) manual test just by executing `CKEDITOR.instances.inline.destroy()`.

Actually I have figured out a better selector that will handle it, without adding too much of an overhead. I'll include it in the review.